src/tables/math.rs

```diff
@@ -0,0 +1,1062 @@
+        let glyph_assembly = s.parse_at_offset16(data);
+        let variant_count = s.read()?;
+        let variants = s.read_array16(variant_count)?;
+        Some(MathGlyphConstruction { glyph_assembly, variants })
+    }
+}
+
+/// Description of math glyph variants.
+#[derive(Debug, Default, Copy, Clone)]
+pub struct MathGlyphVariantRecord {
+    /// Glyph ID for the variant.
+    pub variant_glyph: GlyphId,
+    /// Advance width/height, in design units, of the variant, in the direction of requested glyph
+    /// extension.
+    pub advance_measurement: u16,
+}
+
+impl FromData for MathGlyphVariantRecord {
+    const SIZE: usize = 4;
+    fn parse(data: &[u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        let variant_glyph = s.read()?;
+        let advance_measurement = s.read()?;
+        Some(MathGlyphVariantRecord { variant_glyph, advance_measurement })
+    }
+}
+
+/// How the shape for this glyph can be assembled from parts found in the glyph set of the font.
+#[allow(missing_debug_implementations)]
+#[derive(Clone)]
+pub struct GlyphAssembly<'a> {
+    /// Italics correction of this GlyphAssembly. Should not depend on the assembly size.
+    pub italics_correction: MathValueRecord<'a>,
+    /// Array of part records, from left to right (for assemblies that extend horizontally) or
+    /// bottom to top (for assemblies that extend vertically).
+    pub glyph_parts: LazyArray16<'a, GlyphPartRecord>,
+}
+
+impl<'a> FromSlice<'a> for GlyphAssembly<'a> {
+    fn parse(data: &'a [u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        let italics_correction = s.read::<RawMathValueRecord>()?.get(data);
+        let glyph_part_count = s.read()?;
+        let glyph_parts = s.read_array16(glyph_part_count)?;
+        Some(GlyphAssembly { italics_correction, glyph_parts })
+    }
+}
+
+/// Description of glyph parts to be assembled.
+#[allow(missing_debug_implementations)]
+#[derive(Copy, Clone)]
+pub struct GlyphPartRecord {
+    /// Glyph ID for the part.
+    pub glyph_id: GlyphId,
+    /// Lengths of the connectors on the start of the glyph.
+    pub start_connector_length: u16,
+    /// Lengths of the connectors on the end of the glyph.
+    pub end_connector_length: u16,
+    /// The full advance of the part. It is also used to determine the measurement of the result
+    /// by using the following formula:
+    ///
+    /// _Size of Assembly = Offset of the Last Part + Full Advance of the Last Part_
+    pub full_advance: u16,
+}
+
+impl FromData for GlyphPartRecord {
+    const SIZE: usize = 8;
+    fn parse(data: &[u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        Some(GlyphPartRecord {
+            glyph_id: s.read()?,
+            start_connector_length: s.read()?,
+            end_connector_length: s.read()?,
+            full_advance: s.read()?,
+        })
+    }
+}
+
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug)]
+pub struct PartFlags(pub u16);
```

Comment:
This type should not be public, afaik.

Reply:
That was my fault. This type is intended to be used in the `GlyphPartRecord`, but somehow I forgot about it and left it unused. Now there is a `part_flags` field with this type.